contentcuration/contentcuration/viewsets/tree.py

```diff
@@ -77,55 +67,69 @@ def id_attr(cls):
 
     def list(self, request, *args, **kwargs):
         channel_id = request.query_params.get("channel_id")
-        if channel_id is None:
+        tree_id = request.query_params.get("tree_id")
+
+        if channel_id is None and tree_id is None:
             raise MissingRequiredParamsException(
-                "channel_id query parameter is required but was missing from the request"
+                "tree_id or channel_id query parameter is required but was missing from the request"
             )
-        root = get_object_or_404(ContentNode, channel_main=channel_id)
+
+        root_filter = dict()
+        if channel_id is not None:
+            root_filter.update(channel_main=channel_id)
+        else:
+            root_filter.update(pk=tree_id)
+
+        root = get_object_or_404(ContentNode, **root_filter)
+
+        if tree_id is None:
+            tree_id = root.pk
 
         def map_data(item):
             item["sort_order"] = item.pop("lft")
             item["channel_id"] = channel_id
+            item["tree_id"] = tree_id
             return item
 
-        queryset = self.filter_queryset(root.get_descendants())
+        queryset = self.filter_queryset(root.get_descendants(include_self=True))
         tree = map(map_data, queryset.values(*self.values))
         return Response(tree)
 
-    def move(self, pk, *args, **kwargs):
-        try:
-            contentnode = ContentNode.objects.get(pk=pk)
-        except ContentNode.DoesNotExist:
-            error = ValidationError("Specified node does not exist")
-            return str(error), None
-        target = kwargs.pop("target", None)
-        position = kwargs.pop("position", "first-child")
+
+def move(self, pk, *args, **kwargs):
+    try:
```

Comment:
I have refactored this quite heavily in my PR https://github.com/learningequality/studio/pull/1765, so we will likely get a conflict here - if I am reading it correctly, this is just making it an independent function?

The main focus of my PR was putting this all into the ContentNode queryset manager, so that no matter how these things are called, it gets properly locked.

Reply:
Yeah this was an accident somehow. I've reverted it.

Reply:
And there will be major conflicts, no getting around it :)

Reply:
My PR only touched 100 files!